elasticsearch/src/main/scala/com/snowplowanalytics/stream/loader/ElasticsearchLoader.scala

```diff
@@ -61,7 +61,7 @@ object ElasticsearchLoader {
     config.input match {
       case Source.Stdin      => System.exit(1)
       case _: Source.Kinesis => System.exit(1)
-      case _: Source.Nsq     => ()
+      case _: Source.Nsq     => sys.addShutdownHook(executor.close())
```

Comment:
I don't get why we need to do this only for NSQ?

Reply:
The answer is in the comments above it.  It's because `executor.run()` (line 55) is blocking on the kinesis/stdin apps.  So for kinesis/stdin the execution flow never gets as far as this line of code.

Reply:
Oops sorry the comment was not appearing in Github UI, I should have checked